src/databricks/labs/ucx/source_code/linters/spark_connect.py

```diff
@@ -238,6 +240,10 @@ def lint(self, node: NodeNG) -> Iterator[Advice]:
 
 class SparkConnectLinter(PythonLinter):
     def __init__(self, session_state: CurrentSessionState):
+        if session_state.data_security_mode != DataSecurityMode.USER_ISOLATION:
```

Comment:
Why do we want to run this only for `USER_ISOLATION` security mode?

Reply:
UC Shared Clusters use Spark connect, this linter detects use of legacy APIs that are not supported  in Spark Connect mode

Reply:
`databricks labs ucx lint-local-code` command sets no data security mode, so these linters would be omitted. 
```suggestion
        if session_state.data_security_mode is not None and session_state.data_security_mode != DataSecurityMode.USER_ISOLATION:
```

Reply:
What's the expectation for `databricks labs ucx lint-local-code`? These errors emitted by `SparkConnectLinter` are relevant for UC Shared Clusters, but they have no use for Single-User ones. While it's still better to migrate off using deprecated APIs, it's not possible to achieve for all workloads (e.g. using RDD APIs on a UC SIngle User cluster). I wonder if we need to pass `data_security_mode` to `lint-local-code` as an optional flag